lib/mixlib/install/backend/base.rb

```diff
@@ -195,6 +224,43 @@ def normalize_architecture(architecture)
             architecture
           end
         end
+
+        private
+
+        #
+        # Custom map Chef's supported windows desktop versions to the server versions we currently build
+        # See https://docs.chef.io/platforms.html
+        #
+        def map_custom_windows_desktop_versions(desktop_version)
+          unless SUPPORTED_WINDOWS_DESKTOP_VERSIONS.include?(desktop_version)
+            raise UnsupportedVersion, "Unsupported Windows desktop version `#{desktop_version}`. Supported versions: #{SUPPORTED_WINDOWS_DESKTOP_VERSIONS.join(", ")}."
+          end
+
+          server_version = Util.map_windows_desktop_version(desktop_version)
+
+          # Windows desktop 10 officially maps to server 2016.
+          # However, we don't test on server 2016 at this time, so we default to 2012r2
+          server_version = "2012r2" if server_version == "2016"
+
+          server_version
+        end
+
+        #
+        # Clone all supported Windows desktop artifacts from a base artifact
+        # options hash allows overriding any valid attribute
+        #
+        def clone_windows_desktop_artifacts(base_artifact, options = {})
+          cloned_artifacts = [ ]
+
+          SUPPORTED_WINDOWS_DESKTOP_VERSIONS.each do |dv|
+            options[:platform_version] = dv
+            options[:url] = base_artifact.url.gsub("\/#{base_artifact.platform_version}\/", "\/#{map_custom_windows_desktop_versions(dv)}\/")
+
+            cloned_artifacts << base_artifact.clone_with(options)
+          end
+
+          cloned_artifacts
+        end
```

Comment:
Fwiw you could always use `Enumerable.each_with_object` for your accumulators, like:
```ruby
        def clone_windows_desktop_artifacts(base_artifact, options = {})
          SUPPORTED_WINDOWS_DESKTOP_VERSIONS.each_with_object([]) do |dv, cloned_artifacts|
            options[:platform_version] = dv
            options[:url] = base_artifact.url.gsub("\/#{base_artifact.platform_version}\/", "\/#{map_custom_windows_desktop_versions(dv)}\/")

            cloned_artifacts << base_artifact.clone_with(options)
          end
        end
```
...though I prefer collect (map) instead as it's simpler and more directly suggests the activity, I think. =^)
```ruby
        def clone_windows_desktop_artifacts(base_artifact, options = {})
          SUPPORTED_WINDOWS_DESKTOP_VERSIONS.collect do |dv|
            options[:platform_version] = dv
            options[:url] = base_artifact.url.gsub("\/#{base_artifact.platform_version}\/", "\/#{map_custom_windows_desktop_versions(dv)}\/")

            base_artifact.clone_with(options)
          end
        end
```